range_sensor_layer/src/range_sensor_layer.cpp

```diff
@@ -354,6 +359,21 @@ void RangeSensorLayer::updateCostmap(sensor_msgs::Range& range_message, bool cle
 
   buffered_readings_++;
   last_reading_time_ = ros::Time::now();
+  if(use_decay_)
+    timeCheck();
+}
+
+void RangeSensorLayer::timeCheck()
+{
+  std::map<std::pair<int,int>, double>::iterator it_map;
+  for (it_map = point_map.begin() ; it_map != point_map.end() ; it_map++ )
+  {
+    if(pixel_decay_ < last_reading_time_.toSec() - it_map->second)
```

Comment:
Just for minor efficiency, can you move the conversion of last_reading_time_ to a double outside of the loop?